src/rdkafka_partition.h

```diff
@@ -768,6 +768,11 @@ int rd_kafka_topic_partition_list_find_idx(
     const char *topic,
     int32_t partition);
 
+rd_kafka_topic_partition_t *rd_kafka_topic_partition_list_find_by_id(
```

Comment:
There are many confusing functions for the same. Can we improve the readability of the names and explains what the function is going to return?